From f9591b066ce8647e3c08cbbcdb2a8d0373e05891 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Fri, 25 Jan 2013 11:34:35 +0100
Subject: [PORT FROM MAIN] wifi: Cancel sched scan and scan before enabling pno, enable scan after disabling pno

BZ: 60560

Cancel any pending regular scan or supplicant sched scan (not driver's)
for power consumption optimization.
Reschedule a scan when pno is disabled.

Feature   : CWS.WIFI-Bluewoods

Change-Id: Ie3c62f955876f9f8134a169e2559336307e89397
Orig-Change-Id: I3763bc3736e5abcb90d83d981d60fc05f9599c6a
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
---
 wpa_supplicant/ctrl_iface.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 6335605..b7f271d 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -376,10 +376,16 @@ static int wpa_supplicant_ctrl_iface_set(struct wpa_supplicant *wpa_s,
 		wpa_tdls_enable(wpa_s->wpa, !disabled);
 #endif /* CONFIG_TDLS */
 	} else if (os_strcasecmp(cmd, "pno") == 0) {
-		if (atoi(value))
+		if (atoi(value)) {
+			wpa_supplicant_cancel_sched_scan(wpa_s);
+			wpa_supplicant_cancel_scan(wpa_s);
 			ret = pno_start(wpa_s);
-		else
+		}
+		else {
 			ret = pno_stop(wpa_s);
+			if(wpa_supplicant_req_sched_scan(wpa_s))
+				wpa_supplicant_req_scan(wpa_s, 0, 0);
+		}
 	} else if (os_strcasecmp(cmd, "radio_disabled") == 0) {
 		int disabled = atoi(value);
 		if (wpa_drv_radio_disable(wpa_s, disabled) < 0)
-- 
1.7.4.1

